sdk/helper/ldaputil/client.go

```diff
@@ -478,11 +479,15 @@ func sidBytesToString(b []byte) (string, error) {
 }
 
 func (c *Client) performLdapTokenGroupsSearch(cfg *ConfigEntry, conn Connection, userDN string) ([]*ldap.Entry, error) {
+	var wg sync.WaitGroup
+	var lock sync.Mutex
+	taskChan := make(chan string)
+	maxWorkers := 10
+
 	result, err := conn.Search(&ldap.SearchRequest{
-		BaseDN:       userDN,
-		Scope:        ldap.ScopeBaseObject,
-		DerefAliases: ldapDerefAliasMap[cfg.DerefAliases],
```

Comment:
I think this got lost in the shuffle, perhaps?

Reply:
Really good catch. Removed during debugging and forgot to add it back. Thanks!